File: AdminPanel1/AdminPanel1.Web/Modules/Product/Product/ProductColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace AdminPanel1.Product.Columns;

[ColumnsScript("Product.Product")]
[BasedOnRow(typeof(ProductRow), CheckNames = true)]
public class ProductColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string CatName { get; set; }
    public string Photo { get; set; }
    public string Type { get; set; }
    public string SupplierName { get; set; }
    public DateTime EntryDate { get; set; }
    public string ReviewUrl { get; set; }
    public int Quantity { get; set; }
}

File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/Product/ProductColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductRow } from "./ProductRow";

export interface ProductColumns {
    Id: Column<ProductRow>;
    Name: Column<ProductRow>;
    Description: Column<ProductRow>;
    Price: Column<ProductRow>;
    CatName: Column<ProductRow>;
    Photo: Column<ProductRow>;
    Type: Column<ProductRow>;
    SupplierName: Column<ProductRow>;
    EntryDate: Column<ProductRow>;
    ReviewUrl: Column<ProductRow>;
    Quantity: Column<ProductRow>;
}

export class ProductColumns extends ColumnsBase<ProductRow> {
    static readonly columnsKey = 'Product.Product';
    static readonly Fields = fieldsProxy<ProductColumns>();
}